src/main/java/com/twilio/http/Request.java

```diff
@@ -175,9 +175,24 @@ private String buildURL() {
                 host = joinIgnoreNull(".", product, targetEdge, targetRegion, domain);
             }
 
-            return new URI(parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
-                    parsedUrl.getPath(), parsedUrl.getQuery(), parsedUrl.getRef()).toURL().toString();
-        } catch (final MalformedURLException | URISyntaxException e) {
+            int port = -1;
+            String urlPort = null;
+            if (parsedUrl.getPort() != -1) {
+                port = parsedUrl.getPort();
+                urlPort = ":" + port;
+            }
+            String protocol = parsedUrl.getProtocol() + "://";
+            String path = parsedUrl.getPath().replace("|", "%7C");
```

Comment:
This needs to use a proper encoder. There are many other characters that it would cover.